sw/airborne/modules/computer_vision/opticflow/opticflow_calculator.c

```diff
@@ -528,29 +534,75 @@ bool calc_fast9_lukas_kanade(struct opticflow_t *opticflow, struct image_t *img,
 
   // TODO scale flow to rad/s here
 
+  // ***************
   // Flow Derotation
+  // ***************
+
   float diff_flow_x = 0.f;
   float diff_flow_y = 0.f;
 
   if (opticflow->derotation && result->tracked_cnt > 5) {
-    diff_flow_x = (opticflow->img_gray.eulers.phi - opticflow->prev_img_gray.eulers.phi) * OPTICFLOW_FX;
-    diff_flow_y = (opticflow->img_gray.eulers.theta - opticflow->prev_img_gray.eulers.theta) * OPTICFLOW_FY;
-    /*diff_flow_x = (cam_state->rates.p)  / result->fps * img->w /
-                  OPTICFLOW_FOV_W;// * img->w / OPTICFLOW_FOV_W;
-    diff_flow_y = (cam_state->rates.q) / result->fps * img->h /
-                  OPTICFLOW_FOV_H;// * img->h / OPTICFLOW_FOV_H;*/
+
+    // determine the roll, pitch, yaw differencces between the images.
+    float phi_diff = opticflow->img_gray.eulers.phi - opticflow->prev_img_gray.eulers.phi;
+    float theta_diff = opticflow->img_gray.eulers.theta - opticflow->prev_img_gray.eulers.theta;
+    float psi_diff = opticflow->img_gray.eulers.psi - opticflow->prev_img_gray.eulers.psi;
+
+    if (strcmp(OPTICFLOW_CAMERA.dev_name, "/dev/video0") == 0) {
+      // bottom cam: just subtract a scaled version of the roll and pitch difference from the global flow vector:
+      diff_flow_x = phi_diff * OPTICFLOW_FX;
+      diff_flow_y = theta_diff * OPTICFLOW_FY;
```

Comment:
These parameters now look redundant, maybe you can reuse the parameters from the new struct you added. Otherwise people might end up setting the wrong parameters in their airframe. For this, you will have to update the variable names in the airframes that use the old parameter with the new one and update the documentation of the optical flow module.

Reply:
FX and FY are indeed redundant, but not the correction factor. Also, all airframe files that use OPTICFLOW_FX and OPTICFLOW_FY would have to be changed. The current idea in the pull request is to keep handling the bottom cam as was (since this worked), and just to add a method to derotate flow for the frontal cam.

Reply:
PS: will in any case have a look at it